bytepushers-common-js-package/src/main/javascript/com.bytepushers.base.app.js

```diff
@@ -127,6 +127,263 @@
+    };
+
+    /**
+     * <p>Static function that tells you whether a date is the last day in a month or not.</p>
+     * @private
+     * @param {<a href="http://www.w3schools.com/jsref/jsref_obj_date.asp">Number</a>} time Represents the time you want to add to the date.
+     *
+     * @returns {String} A new Date object with the specified time added.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.isLastDayInMonth = function (date) {
+        "use strict";
+        var lastDayInMonth = date.getCurrentMonthTotalDays();
+        if (date.getDate() === lastDayInMonth) {
+            return true;
+        }
+        return false;
+    };
+
+    /**
+     * <p>Static field that is used to get calendar total calendar days of the previous month.</p>
+     * @static
+     * @function
+     * @returns {@link <a href="http://www.w3schools.com/jsref/jsref_obj_number.asp">Number</a>} The total days in the previous month.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.prototype.getPreviousMonthTotalDays = function () {
+        "use strict";
+        if (this.getMonth() === 0) {
+            return Date.monthNames[11].getTotalDays(this.getFullYear());
+        } else {
+            return Date.monthNames[this.getMonth() - 1].getTotalDays(this.getFullYear());
+        }
+    };
+
+    /**
+     * <p>Static function that is used to get the total calendar days of the next month.</p>
+     * @static
+     * @function
+     * @returns {@link <a href="http://www.w3schools.com/jsref/jsref_obj_number.asp">Number</a>} The total days in the next month.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.prototype.getNextMonthTotalDays = function () {
+        "use strict";
+        if (this.getMonth() === 11) {
+            return Date.monthNames[0].getTotalDays(this.getFullYear());
+        } else {
+            return Date.monthNames[this.getMonth() + 1].getTotalDays(this.getFullYear());
+        }
+    };
+
+    /**
+     * <p>Function that is used to get the total calendar days of the next month.</p>
+     * @static
+     * @function
+     * @returns {@link <a href="http://www.w3schools.com/jsref/jsref_obj_number.asp">Number</a>} The total days in the next month.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.prototype.getCurrentMonthTotalDays = function () {
+        "use strict";
+        if (this.getMonth() === 11) {
+            return Date.monthNames[0].getTotalDays(this.getFullYear());
+        } else {
+            return Date.monthNames[this.getMonth()].getTotalDays(this.getFullYear());
+        }
+    };
+
+    /**
+     * <p>Adds time to a date object.</p>
+     * @private
+     * @param {<a href="http://www.w3schools.com/jsref/jsref_obj_number.asp">Number</a>} time Represents the time you want to add to the date.
```

Comment:
Added @static annotation since this method is a static method.
And remove unnecessary space.